src/shims/unix/fd.rs

```diff
@@ -619,22 +627,68 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let bytes = this.read_bytes_ptr_strip_provenance(buf, Size::from_bytes(count))?.to_owned();
         // We temporarily dup the FD to be able to retain mutable access to `this`.
         let Some(fd) = this.machine.fds.get(fd) else {
-            return Ok(Scalar::from_target_isize(this.fd_not_found()?, this));
+            let res = this.fd_not_found()?;
+            this.write_scalar(Scalar::from_target_isize(res, this), dest)?;
+            return Ok(());
         };
 
-        let result = match offset {
-            None => fd.write(&fd, communicate, &bytes, this),
+        let _result = match offset {
```

Comment:
The reason you're getting a bogus message about undef bytes is that you're throwing away the unsupported error here. Thus the return integer is never written, so it's undef and immediately fails with an undef error when read for validation

Reply:
Oh right, this makes sense, thanks!